baseline/tf/embeddings.py

```diff
@@ -26,49 +24,18 @@ def detached_ref(self):
 
         :return:
         """
-        if getattr(self.embedding_layer, '_weights', None) is not None:
-            return type(self)(self._name, weights=self.embedding_layer._weights, **self._state)
-        if hasattr(self.embedding_layer, 'embed') and getattr(self.embedding_layer.init_embed, '_weights') is not None:
-            return type(self)(self._name, weights=self.embedding_layer.init_embed._weights, **self._state)
+        if getattr(self, '_weights', None) is not None:
+            return type(self)(name=self.name, weights=self._weights, **self._state)
+        if hasattr(self, 'embed') and getattr(self.init_embed, '_weights') is not None:
+            return type(self)(name=self.name, weights=self.init_embed._weights, **self._state)
         raise Exception('You must initialize `weights` in order to use this method')
 
-    def get_dsz(self):
-        """Get the number of output dimension of this operation
-
-        :return:
-        """
-        return self.embedding_layer.get_dsz()
-
-    def get_weights(self):
-        return self.embedding_layer.get_weights()
-
-    @property
-    def output_dim(self):
-        return self.embedding_layer.output_dim
-
-    def get_vsz(self):
-        """Get the number of words (including <PAD>) in the vocabulary
-
-        :return:
-        """
-        return self.embedding_layer.get_vsz()
-
-    def encode(self, x):
-        """This instantiates the sub-graph for this object and returns the output node
-
-        :return:
-        """
+    def call(self, x):
         if x is None:
-            x = self.create_placeholder(self._name)
+            x = self.create_placeholder(self.name)
         self.x = x
-        return self.embedding_layer(x)
-
-    def call(self, x):
-        return self.encode(x)
 
-    def get_feed_dict(self):
-        """Return a feed dict that is needed to initialize this embeddings."""
-        return self.embedding_layer.get_feed_dict()
+        return super().encode(x)
```

Comment:
I thought part of the plan of simplifying the embedding was to remove the `encode` and replace it with `call`?